### ISSUE TITLE
Calls remaing pending

### ISSUE THREAD

**Issue by maxxer** [Mancy/Zammad_3CX#3]
Even if the calls are ended on 3cx I see them as pending on Zammad.

Am I missing permission or something?
thanks

![immagine](https://user-images.githubusercontent.com/240201/61464473-99053080-a976-11e9-84f1-dcbbcbf5e972.png)

**Comment by sgi-ch** [comment]
same here.

**Comment by Mancy** [comment]
Actually, the 3CX CRM integration tool does not give a unique identifier to match the endCall request exactly with the answerdCall request, for this reason some calls remain pending, that's why i had to do a work around using the call start timestamp, which does not perfectly match also in every situation. 

Currently 3CX supports CRM lookup and some other options using the Call Flow Designer tool, which I will use, and if possible I will put it in github, 

The Call Flow Designer is available for the standard/free version of 3CX which is good if compared to the poor CRM Template thing.

**Comment by sgi-ch** [comment]
Thanks for the feedback @Mancy. If you need any support or help (dev or testing) please let us know. We've a Professional Version here.

**Comment by maxxer** [comment]
If you can share the *Call Flow Designer* it would be great! Thanks!!

**Comment by sgi-ch** [comment]
this one? https://downloads.3cx.com/downloads/3CXCallFlowDesigner.exe

**Comment by Mancy** [comment]
> this one? https://downloads.3cx.com/downloads/3CXCallFlowDesigner.exe

Yes that's it. please make sure you have latest versions of both the Phone system and the Call Flow Designer

**Comment by sgi-ch** [comment]
Yes, we're up-to-date on both.

**Comment by maxmaury** [comment]
Hi, i'm follow up on this issue. Do you have a solution to fix it or share the call flow project ?
Thanks a lot !

Kind Regards,
Max

**Comment by mhonig** [comment]
I would also be interested to know if there was any progress made as I am facing the same issue. Thank you!

**Comment by herzkerl** [comment]
Hi there! I’ve successfully used (parts of) this script _and_ Call Flow Designer to make it work.
👉🏻 https://community.zammad.org/t/cti-api-integration-with-3cx/1110/32
